src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/DisableRuntimeMarshalling.cs

```diff
@@ -127,32 +138,237 @@ internal sealed class DisableRuntimeMarshallingAnalyzer : DiagnosticAnalyzer
+                        }
+                    }
+
+                    _cache.TryAdd(type, false);
+                    return false;
+                }
+            }
+        }
+
+        private class DelegateInteropUsageAnalyzer
+        {
+            private readonly IMethodSymbol? _getDelegateForFunctionPointerNonGeneric;
+            private readonly IMethodSymbol? _getDelegateForFunctionPointerGeneric;
+            private readonly IMethodSymbol? _getFunctionPointerForDelegateNonGeneric;
+            private readonly IMethodSymbol? _getFunctionPointerForDelegateGeneric;
+            private readonly AutoLayoutTypeCache _autoLayoutCache;
+
+            public DelegateInteropUsageAnalyzer(Compilation compilation, AutoLayoutTypeCache autoLayoutCache)
+            {
+                if (compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesMarshal, out var marshalType))
+                {
+                    var getDelegateForFunctionPointerMethods = marshalType.GetMembers("GetDelegateForFunctionPointer").OfType<IMethodSymbol>();
+                    _getDelegateForFunctionPointerNonGeneric = getDelegateForFunctionPointerMethods.First(m => m.TypeArguments.Length == 0);
+                    _getDelegateForFunctionPointerGeneric = getDelegateForFunctionPointerMethods.First(m => m.TypeArguments.Length != 0);
+                    var getFunctionPointerForDelegateMethods = marshalType.GetMembers("GetFunctionPointerForDelegate").OfType<IMethodSymbol>();
+                    _getFunctionPointerForDelegateNonGeneric = getFunctionPointerForDelegateMethods.First(m => m.TypeArguments.Length == 0);
+                    _getFunctionPointerForDelegateGeneric = getFunctionPointerForDelegateMethods.First(m => m.TypeArguments.Length != 0);
+                }
+
+                _autoLayoutCache = autoLayoutCache;
+            }
+
+            public void RegisterActions(CompilationStartAnalysisContext context, bool hasDisableRuntimeMarshallingAttribute)
+            {
+                context.RegisterOperationAction(AnalyzeInvocation, OperationKind.Invocation);
+
+                if (!hasDisableRuntimeMarshallingAttribute)
+                {
+                    context.RegisterSymbolAction(AnalyzeMethod, SymbolKind.Method);
+
+                    context.RegisterOperationAction(AnalyzeLocalFunction, OperationKind.LocalFunction);
+                }
+            }
+
+            private void AnalyzeInvocation(OperationAnalysisContext context)
+            {
+                IInvocationOperation operation = (IInvocationOperation)context.Operation;
+                if (SymbolEqualityComparer.Default.Equals(operation.TargetMethod.OriginalDefinition, _getDelegateForFunctionPointerNonGeneric))
+                {
+                    if (operation.Arguments.Length == 2 && operation.Arguments[1] is { Value: ITypeOfOperation { TypeOperand: INamedTypeSymbol { TypeKind: TypeKind.Delegate } delegateType } })
```

Comment:
I think you want to use the below extension method to get the correct argument: https://github.com/dotnet/roslyn-analyzers/blob/04956180bcaac015cffa8f80e4836325e1033f80/src/Utilities/Compiler/Extensions/IOperationExtensions.cs#L823-L825

Note that `operation.Arguments[1]` will always give you the second argument, so if named arguments are used where your callsite is `MethodCall(param2: value2, param1: value1);`, than `operation.Arguments[1]` will give you `param1: value1`, while you want `param2: value2`. It would be good to also add a test for this case.